lib/fields.ts

```diff
@@ -0,0 +1,70 @@
+// Helper functions for creating fields
+
+import { FieldDefinition, ParquetType, SchemaDefinition } from "./declare";
+
+export function createStringField(optional = true, fieldOptions: FieldDefinition = {}): FieldDefinition {
+    return { ...fieldOptions, optional, type: 'UTF8' };
+}
+
+export function createBooleanField(optional = true, fieldOptions: FieldDefinition = {}): FieldDefinition {
+    return { ...fieldOptions, optional, type: 'BOOLEAN' };
+}
+
+export function createIntField(size: 32 | 64, optional = true, fieldOptions: FieldDefinition = {}): FieldDefinition {
+    return { ...fieldOptions, optional, type: `INT${size}` };
+}
+
+export function createFloatField(optional = true, fieldOptions: FieldDefinition = {}): FieldDefinition {
+    return { ...fieldOptions, optional, type: 'FLOAT' };
+}
+
+export function createDoubleField(optional = true, fieldOptions: FieldDefinition = {}): FieldDefinition {
+    return { ...fieldOptions, optional, type: 'DOUBLE' };
+}
+
+export function createDecimalField(precision: number, optional = true, fieldOptions: FieldDefinition = {}): FieldDefinition {
+    return { ...fieldOptions, precision, optional, type: 'FLOAT' };
+}
+
+export function createTimestampField(optional = true, fieldOptions: FieldDefinition = {}): FieldDefinition {
+    return { ...fieldOptions, optional, type: 'TIMESTAMP_MILLIS' };
+}
+
+export function createStructField(fields: SchemaDefinition, optional = true): FieldDefinition {
+    return {
+        optional,
+        fields,
+    }
+}
+
+export function createStructListField(fields: SchemaDefinition, optional = true): FieldDefinition {
+    return {
```

Comment:
I found some time to test the changes locally.

It seems to work and the parquet file is generated successfully, but if I try to open the generated parquet file in an external parquet viewer ( in my case via this [PyCharm Plugin](https://github.com/benwatson528/intellij-avro-parquet-plugin) ) I got the following error message:

```
Invalid list type repeated group struct_field (LIST) {
  required group list {
    optional binary sub3 (STRING);
    optional binary sub4 (STRING);
    optional group sub5 {
      optional binary sub6 (STRING);
      optional binary sub7 (STRING);
    }
    optional group sub8 (LIST) {
      repeated group list {
        required binary element (STRING);
      }
    }
  }
}


```

If I change the return value to:

```js
  return {
    type: "LIST",
    optional,
    fields: {
      list: {
        repeated: true,
        fields,
      },
    },
  }
```

The error doesn't occur.